src/libraries/System.Private.CoreLib/src/System/Exception.cs

```diff
@@ -86,7 +87,9 @@ public virtual Exception GetBaseException()
 
         public virtual string? Source
         {
-            get => _source ??= CreateSourceName();
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+                Justification = "The API will return <unknown> if the metadata for current method cannot be established.")]
+            get => _source ??= HasBeenThrown ? (TargetSite?.Module.Assembly.GetName().Name ?? "<unknown>") : null;
```

Comment:
Can `TargetSite` be `null`? If so, perhaps "<unknown>" should be returned for that as well?

Reply:
Yes, `TargetSite` can be null in some corner cases and the code as written will return `<unknown>` for that.